src/pages/docs/rpc.mdx

```diff
@@ -0,0 +1,240 @@
+
+Golem instead introduces a new way to do worker-to-worker communication, using a method called **WASM RPC**. This consists of the following building blocks:
+
+- Golem provides a **host resource** for workers, which can be imported into any Golem project and used to invoke a remote worker by passing the target worker, the name of the invoked function and its parameters in a dynamic way using a data type called `wit-value`.
+- A CLI tool called `wasm-rpc-stubgen` can take any Golem worker, and generate a pair of a WIT definition and a WASM component implementing a "stub" for remotely calling that given worker. The generated stub internally uses the above mentioned low-level invocation API, providing a typed interface for working with a remote Golem worker.
+- The generated stub can be **composed** with other Golem workers using standard WASM tooling to get a resulting WASM ready to be uploaded as a Golem template, capable of typed worker-to-worker communication.
+
+## Details
+
+### Low-level interface
+
+The low-level host interface this technique uses is defined in [wasm-rpc.wit](https://github.com/golemcloud/golem-wit/blob/main/wit/deps/wasm-rpc/wasm-rpc.wit) as the following:
+
+```rust
+package golem:rpc@0.1.0;
+
+interface types {
+  type node-index = s32;
+
+  record wit-value {
+    nodes: list<wit-node>,
+  }
+
+  variant wit-node {
+    record-value(list<node-index>),
+    variant-value(tuple<u32, option<node-index>>),
+    enum-value(u32),
+    flags-value(list<bool>),
+    tuple-value(list<node-index>),
+    list-value(list<node-index>),
+    option-value(option<node-index>),
+    result-value(result<option<node-index>, option<node-index>>),
+    prim-u8(u8),
+    prim-u16(u16),
+    prim-u32(u32),
+    prim-u64(u64),
+    prim-s8(s8),
+    prim-s16(s16),
+    prim-s32(s32),
+    prim-s64(s64),
+    prim-float32(float32),
+    prim-float64(float64),
+    prim-char(char),
+    prim-bool(bool),
+    prim-string(string),
+  }
+
+  record uri {
+    value: string,
+  }
+
+  variant rpc-error {
+    protocol-error(string),
+    denied(string),
+    not-found(string),
+    remote-internal-error(string)
+  }
+
+  resource wasm-rpc {
+    constructor(location: uri);
+
+    invoke-and-await: func(function-name: string, function-params: list<wit-value>) -> result<wit-value, rpc-error>;
+  }
+}
+
+world wit-value {
+    import types;
+}
+```
+
+### Worker URI
+
+The **worker URI** that has to be passed to the remote interface as a parameter identifies a specific **Golem worker**, consisting of the **template ID** and the **worker name**.
+It can be constructed by hand using the pattern `worker://template-id/worker-name`, and a running worker can get it's own URI using the Golem-specific host function `golem:api/host/get-self-uri`.
```

Comment:
```suggestion
It can be constructed by hand using the pattern `worker://template-id/worker-name`, and a running worker can get its own URI using the Golem-specific host function `golem:api/host/get-self-uri`.
```